src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -272,16 +273,23 @@ public override void UpdateAssetCache()
                 unvisitedAssets.Remove(asset);
                 if (!oldAssets.ContainsKey(asset))
                 {
-                    oldAssets.Add(asset, new TAssetEntry
+                    if (AssetDatabase.TryGetGUIDAndLocalFileIdentifier(asset, out string guid, out long localid))
                     {
-                        AssetId = Guid.NewGuid(),
-                        Asset = asset
-                    });
+                        oldAssets.Add(asset, new TAssetEntry
+                        {
+                            AssetId = new AssetId(new Guid(guid), localid),
+                            Asset = asset
+                        });
+                    }
+                    else
+                    {
+                        Debug.LogError($"Unable to obtain id for Asset: {asset.ToString()}");
+                    }
                 }
             }
 
             CleanUpUnused(oldAssets, unvisitedAssets);
-            assets = oldAssets.Values.ToArray();
+            assets = oldAssets.Values.OrderBy(x => x.AssetId.FileIdentifier).ThenBy(x => x.AssetId.Guid).ToArray();

```

Comment:
>OrderBy(x => x.AssetId.FileIdentifier).ThenBy(x => x.AssetId.Guid) [](start = 38, length = 66)

If we're changing the mechanics of how our assets are enumerated (i.e. the EnumerateAllAssets). it might be worth documenting in the surfaces where this is exposed (i.e. EnumerateAllAssets will enumerate assets on File Identifier and then Guid)

Reply:
Wouldn't we want to orderby Guid first, then file Id? As Guid represents the file, and fileId is the id within that file